sockeye/inference.py

```diff
@@ -579,6 +580,17 @@ def chunks(self, chunk_size: int) -> Generator['TranslatorInput', None, None]:
                                   chunk_id=chunk_id)
 
 
+class BadTranslatorInput(TranslatorInput):
+
+    def __init__(self, sentence_id, tokens):
+        super().__init__(sentence_id=sentence_id, tokens=tokens, chunk_id=-1, factors=None)
+
+
+def _bad_input(sentence_id: int) -> BadTranslatorInput:
+    logger.warning("Bad input (%d) will return empty output.", sentence_id)
```

Comment:
it would probably make sense to log the actual input rather than the sentence id, so that one could rerun a bad input from the log file.

Reply:
I am doing this in the individual functions because its hard to decide what the logged string should be (json, multiple stirngs, factored string).

Reply:
make_input_from_json_string actually does not log the translation as far as I can see. Alternatively could have a field `BadTranslatorInput.reason` where you fill in a string from each one of the methods and then just do the logging of the entire input string along with the reason one level up.